### ISSUE TITLE
Eslint doesn't detect globally injected types

### ISSUE THREAD

**Issue by alex-grover** [ota-meshi/eslint-plugin-astro#96]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

v8.22.0

### What version of `eslint-plugin-astro` are you using?

v0.17.1

### What did you do?

<details>
<summary>Configuration</summary>

```json
"eslintConfig": {
  "extends": [
    "eslint:recommended",
    "plugin:astro/recommended"
  ],
  "overrides": [
    {
      "files": [
        "*.astro"
      ],
      "parser": "astro-eslint-parser",
      "parserOptions": {
        "parser": "@typescript-eslint/parser",
        "extraFileExtensions": [
          ".astro"
        ]
      }
    }
  ]
}
```
</details>

Trying to lint the `.astro` file below, which is taken from the [blog template project](https://github.com/withastro/astro/blob/d54588c7a4adfa05969713111d36673f3a9b988e/examples/blog/src/components/HeaderLink.astro). The only modification I made was to add the `/// <reference types="astro/astro-jsx" />` on the second line, as recommended by [the docs](https://docs.astro.build/en/guides/typescript/#built-in-attribute-types).

```astro
---
/// <reference types="astro/astro-jsx" />

export interface Props extends astroHTML.JSX.AnchorHTMLAttributes {}

const { href, class: className, ...props } = Astro.props as Props
const isActive = href === Astro.url.pathname
---

<a href={href} class:list={[className, { active: isActive }]} {...props}>
  <slot />
</a>
<style>
  a {
    display: inline-block;
    text-decoration: none;
  }
  a.active {
    font-weight: bolder;
    text-decoration: underline;
  }
</style>
```

### What did you expect to happen?

The linter works without warnings or errors

### What actually happened?

```
$ npx eslint src/components/HeaderLink.astro

./src/components/HeaderLink.astro
  4:32  error  'astroHTML' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

### Link to Minimal Reproducible Example

https://github.com/alex-grover/eslint-plugin-astro-types-bug

clone, install, and run `npx eslint src/components/HeaderLink.astro`

**Comment by ota-meshi** [comment]
Turning off the no-undef rule is recommended when using typescript and eslint.
See the following link for details.

https://typescript-eslint.io/docs/linting/troubleshooting#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors

**Comment by alex-grover** [comment]
Got it, thanks! Perhaps it would be worthwhile to bake more default rules into this package (or maybe `astro check`) so it's comprehensive if users don't dig into the configuration? 

Or potentially having a separate config that also sets up TS & JS to work out of the box, as there are other lint issues raised on the starter projects. That's the approach that NextJS takes and probably most people using Astro would expect a similar experience.

**Comment by ota-meshi** [comment]
Your suggestion sounds good to me. This eslint-plugin project is still new, so there is still a lot to do.
But I'm the only one maintaining this project. I need your contribution or your new project

**Comment by chshouyu** [comment]
```json
{
  "globals": {
    "astroHTML": true
  }
}
```

I tried this and it works ok.
